ddtrace/internal/coverage/lines.py

```diff
@@ -18,28 +28,30 @@ def __repr__(self):
         return f"CoverageLines(num_lines={self._num_lines()})"
 
     def _num_lines(self) -> int:
-        return sum([bin(byte).count("1") for byte in self._lines])
+        return sum(_bit_count(byte) for byte in self._lines)
 
     def add(self, line_number: int):
         lines_byte = line_number // 8
-        lines_bit = 2 ** (line_number % 8)
 
         if lines_byte >= len(self._lines):
             self._lines.extend(bytearray(lines_byte - len(self._lines) + 1))
 
+        # DEV this fun bit allows us to trick ourselves into little-endianness, which is what the backend wants to see
+        # in bytes
+        lines_bit = 2 ** (7 - (line_number % 8))
         self._lines[lines_byte] |= lines_bit
 
     def add_lines(self, lines: list[int]):
         for line in lines:
             self.add(line)
 
-    def to_string(self) -> str:
-        return "".join([str(n) for n in self.to_list()])
+    def to_bytes_string(self) -> str:
+        return " ".join(bin(byte)[2:].zfill(8) for byte in self._lines)
 
     def to_list(self) -> list[int]:
         lines = []
         for _byte in self._lines:
-            for _bit in range(8):
+            for _bit in range(7, -1, -1):
                 if _byte & (1 << _bit):
```

Comment:
We can also get rid of this trick if we do it with `>>`:
```suggestion
            for _bit in range(8):
                if _byte & (0b10000000 >> _bit):
```